src/pages/GetStarted/CustomWorkspaceTab/StorageType/__tests__/index.spec.tsx

```diff
@@ -0,0 +1,179 @@
+
+    // perform click on toggle button to show list of options
+    const optionToggleElement = selector.getByRole('button', { name: toTitle(defaultStorageType) });
+    fireEvent.click(optionToggleElement);
+
+    const options = selector.getAllByRole('option');
+
+    expect(options.length).toEqual(3);
+
+    expect(options[0].innerHTML).toMatch(toTitle('async'));
+    expect(options[1].innerHTML).toMatch(toTitle('ephemeral'));
+    expect(options[2].innerHTML).toMatch(toTitle('persistent'));
+  });
+
+  it('should switch to another storage type by choosing corresponding option', () => {
+    const defaultStorageType: che.WorkspaceStorageType = 'persistent';
+    const nextStorageType: che.WorkspaceStorageType = 'ephemeral';
+    const selector = renderSelector(defaultStorageType);
+
+    // click on toggle button to expand the list of options
+    const optionToggleElement = selector.getByRole('button', { name: toTitle(defaultStorageType) });
+    fireEvent.click(optionToggleElement);
+
+    // click on option element to choose it
+    const nextOptionElement = selector.getByText(toTitle(nextStorageType));
+    expect(nextOptionElement).toBeVisible();
+    fireEvent.click(nextOptionElement);
+
+    const nextToggleElement = selector.getByRole('button', { name: toTitle(nextStorageType) });
+    expect(nextToggleElement).toBeVisible();
+
+    expect(mockOnChange).toHaveBeenCalledWith(nextStorageType);
+  });
+
+  it('should handle storage type property changing', () => {
+    const defaultStorageType: che.WorkspaceStorageType = 'persistent';
+    const nextStorageType: che.WorkspaceStorageType = 'ephemeral';
+    const selector = renderSelector(defaultStorageType);
+
+    const store = createFakeStore(defaultStorageType);
+    selector.rerender(
+      <Provider store={store}>
+        <StorageTypeFormGroup
+          storageType={nextStorageType}
+          onChange={mockOnChange}
+        />
+      </Provider>
+    );
+
+    const nextToggleElement = selector.getByRole('button', { name: toTitle(nextStorageType) });
+    expect(nextToggleElement).toBeVisible();
+
+    expect(mockOnChange).toHaveBeenCalledWith(nextStorageType);
+  });
+
+  it('should open modal box with descriptions', () => {
+    const defaultStorageType: che.WorkspaceStorageType = 'persistent';
+    const selector = renderSelector(defaultStorageType);
+
+    // click to open modal dialog
+    const linkButton = selector.getByRole('button', { name: /learn more/i });
+    expect(linkButton).toBeVisible();
+    fireEvent.click(linkButton);
+
+    const modalBox = selector.getByRole('dialog');
+    expect(modalBox).toBeVisible();
+
+    // link to docs
+    const link = selector.getByRole('link', { name: /open documentation/i });
+    expect(link).toBeVisible();
+    expect(link.getAttribute('href')).toEqual('https://che-docs/storage-types');
+  });
+
+});
+
+function createFakeStore(storageType: che.WorkspaceStorageType = 'ephemeral'): Store {
```

Comment:
We already have '/services/__mocks__/store'. Why don't you use it?

Reply:
This is a Draft PR and it's not ready for review yet.

Actually, I implemented class [FakeStoreBuilder](https://github.com/che-incubator/che-dashboard-next/pull/83/files#diff-b080c4cfb111fb040efee14f6a6730f7a9cb102490869d8154664ca19073f5a6) to simplify mocking the store and I'm going to replace all usages of `createFakeStore` with `FakeStorageBuilder`.